client/extensions/zoninator/components/search-autocomplete/suggestions.jsx

```diff
@@ -0,0 +1,156 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { find, findIndex } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import QueryPosts from 'components/data/query-posts';
+import { getSitePostsForQuery } from 'state/posts/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import SuggestionItem from './suggestion-item';
+
+class Suggestions extends Component {
+
+	static propTypes = {
+		suggest: PropTypes.func.isRequired,
+		searchTerm: PropTypes.string,
+		suggestions: PropTypes.array,
+	}
+
+	static defaultProps = {
+		suggestions: [],
+		searchTerm: '',
+	}
+
+	state = {
+		suggestionPosition: 0,
+		currentSuggestion: null,
+	}
+
+	countSuggestions() {
+		return this.props.suggestions ? this.props.suggestions.length : 0;
+	}
+
+	filterSuggestions() {
+		if ( ! this.countSuggestions() ) {
+			return [];
+		}
+
+		const results = this.props.suggestions.filter( ( { slug } ) => ! find( this.props.ignored, { slug } ) );
+
+		return results;
+	}
+
+	getSuggestionForPosition( position ) {
+		return this.props.suggestions[ position ];
```

Comment:
Nah, that's my bad then. This is fine so long as the calling function accounts for the `undefined` possibility.